### DISCUSSION TITLE
src package structure?

### DISCUSSION THREAD

**Discussion by iyanmv** [mhostetter/galois#399] [General]
Hi @mhostetter 

What do you think about reorganizing the repo using a src structure? There are a few good reasons to do so.
Here are some refs that I found useful about this topic:

https://hynek.me/articles/testing-packaging/
https://github.com/pypa/packaging.python.org/issues/320
https://blog.ionelmc.ro/2014/05/25/python-packaging/#the-structure

Also, if you have some arguments in favor of current non src or flat structure, I would like to hear them.

I could work on a PR if you are interested.

**Comment by mhostetter** [comment]
Hey @iyanmv. I did a lot of reading about this today...

It seems to me the big advantage of using  `src/` is that when running `python3 -m pytest tests/` in the root directory the local `galois/` source code won't be imported (because we'd really prefer to import and test the installed package). Is that your understanding?

It looks like I ran into this a while ago and manually removed the `galois/` folder and then installed the newly-created `.whl`.

https://github.com/mhostetter/galois/blob/0ddcbd75272ff86445be5753f17c927a63d43b75/.github/workflows/test.yml#L115-L132

It seems like the community consensus is around using `src/`, which works for me. I'm all about using the **best thing**™️ . From my reading, it also seems like `pyproject.toml` is preferred over `setup.cfg`. I'll have to do more research, but perhaps that could/should be changed in the future too.

Yeah, if you're up for a PR, that'd be great!

**Comment by mhostetter** [comment]
This was merged in #404.
